tools/validate/module-run/runWasm.js

```diff
@@ -318,14 +318,37 @@ const importObject = {
         dyntype_cmp: () => {
             unimplemented();
         },
-        dyntype_new_object_with_class: () => {
-            throw Error('not implemented: fallback to QuickJS on JS');
-        },
-        dyntype_invoke: () => {
-            throw Error('not implemented: fallback to QuickJS on JS');
+        dyntype_new_object_with_class: (ctx, name, args_array) => {
+            let res = undefined;
+            const str_value = cstringToJsString(name);
+            switch (str_value) {
+                case 'Map': {
+                    res = new Map(...args_array);
+                    break;
+                }
+                case 'Date': {
+                    res = new Date(...args_array);
+                    break;
+                }
+                case 'Promise': {
+                    res = new Promise(...args_array);
+                    break;
+                }
+                default: {
+                    throw Error(`not support new ${str_value} class yet`);
+                }
+            }
```

Comment:
How about: 
``` javascript
ctor = global[cstringToJsString(name)]`
return new ctor(...args_array);
```

Reply:
In nodejs, we can use `global` to find the global ctor, but in chrome, we should use `window` to get the global ctor.
If not take chrome in our consideration, we can use `global`, otherwise, I think a more common way is better.